hoverxref/_static/js/hoverxref.js_t

```diff
@@ -144,6 +138,33 @@ $(document).ready(function() {
             );
         }
     })
+}
+
+
+$(document).ready(function() {
+    // Remove ``title=`` attribute for intersphinx nodes that have hoverxref enabled.
+    // It doesn't make sense the browser shows the default tooltip (browser's built-in)
+    // and immediately after that our tooltip was shown.
+
+    // Support lazy-loading here by switching between on-page-load (calling .tooltipster directly)
+    // or delaying it until after a mouseenter or click event.
+    // On large pages (the use case for lazy-loaded tooltipster) this moves dom manipulation to
+    // on-interaction, which is known to be less performant on small pages (as per tooltipster docs),
+    // but on massive docs pages fixes an otherwise severe page load stall when tooltipster
+    // manipulates the html for every single tooltip at once.
+    {% if hoverxref_tooltip_lazy %}
+    $('.{{ hoverxref_css_class_prefix }}hoverxref.{{ hoverxref_css_class_prefix }}tooltip').one('mouseenter click touchstart tap', function(event) {
+        $(this).removeAttr('title');
+        {# If we're lazy-tipstering, we've used up the click event, so open the modal too with tooltipster('open') #}
+        addTooltip($(this)).tooltipster('open');
```

Comment:
I don't feel strongly here, but this could be cleaned up a small amount by keeping this logic in a central function. That way, both the `click`/etc event and the full page element query would use the same function/logic for displaying the tooltip, instead of having slightly different logic.

Reply:
I agree that it could be cleaned up, but I'm not particularly familiar with the codebase as is, so I didn't feel comfortable making the judgement call on how the differences should be handled. As the code is today, it should behave exactly the same except when the optional lazy feature is toggled, and I think that's a good decision to avoid changing how the code behaves for current users. 

The reason the tooltip has to open separately is because I chose to have the tooltip be created on interaction, which is (as far as I can tell) the only way to do it. 

We can't use something like you would expect from lazy-loaded images because inserting html is relatively slow, and scrolling down would involve lots of work adding unneeded tooltips in lazy mode if I used something like a visibility-observer. 

I can't do it slowly in the background because that's just as expensive and will lead to stumbling page behavior, as well as likely missing cases where a user uses a jumplink far down into the page (which is very common in large sphinx docs). 

I could create a "update box" around each link and cause tooltips to be prepped when the mouse gets closer, but that's fairly complicated and additionally fails if the user tabs to or "teleports" their mouse to a link via accessibility tools, a slow machine, or mobile touch, and I don't see a way to solve that nicely.

So I'm left with the option of simply hijacking the hover/tap/click itself. But if I do so, then I will have to manually forward the click event somehow. 

There are three options for this:
A) manually call tooltipster('open')
B) manually forward the hover event with .trigger()
C) manually call the tooltipster event handler (why not just use A)

B) isn't a good solution because it would make hover/click/tap events double-fire on tooltipstered links, which would interact with any other event handlers on the same links/propagation tree in an unexpected way.

Reply:
I think it probably makes sense to leave it as-is because the current implementation works on our preview, and I'd be concerned about doing a refactor without having the same amount of testing.